lib/GADS/Datum/File.pm

```diff
@@ -153,10 +153,27 @@ sub _build_files
     elsif ($self->has_ids) {
         @return = $self->_ids_to_files(@{$self->ids});
     }
+    
+    return [+{
+        id => -1,
+        name => 'Purged',
+        mimetype => 'text/plain',
+    }] if $self->_is_purged;
 
     return \@return;
 }
 
+has _is_purged => (
+    is=>'lazy',
+    builder => sub {
+        my $self = shift;
+        my $id = $self->record_id;
+        my $datum = $self->schema->resultset('File')->find($id) if $id;
```

Comment:
I don't think this will return the correct datum. I think you will need a new method which returns the applicable `GADS::Schema::Result::File`, e.g.
```
sub _datums
{   my $self = shift;
    $self->schema->resultset('File')->search({
        record_id => $self->record_id,
        layout_id => $self->column->id,
    })->all;
}
```

Then create a new `is_purged` method